pkg/asset/installconfig/azure/session.go

```diff
@@ -224,9 +235,31 @@ func newSessionFromCredentials(cloudEnv azureenv.Environment, credentials *Crede
 		AADEndpoint:  cloudEnv.ActiveDirectoryEndpoint,
 	}
 	c.Resource = cloudEnv.TokenAudience
-	authorizer, err := c.Authorizer()
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to get client credentials authorizer")
+
+	var authorizer autorest.Authorizer
+	switch cloudEnv.Name {
+	case azure.StackCloud.Name():
+		logrus.Debug("Falling back to deprecated ADAL authentication")
+		// The new authorization method is not yet supported for private clouds
+		// See https://github.com/Azure/azure-sdk-for-go/pull/16942
+		var err error
+		authorizer, err = c.Authorizer()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get client credentials authorizer")
+		}
+	default:
```

Comment:
It may take some time to get CI setup on govcloud so I'm testing this manually:

```
FATAL failed to fetch Master Machines: failed to load asset "Install Config": platform.azure.region: Invalid value: "usdodcentral": region "usdodcentral" does not support resource creation 
```

I think we need another switch statement here for govcloud and to pass in govcloud as the authority host as described here (this example uses China, but I believe govcloud should behave similarly): https://github.com/Azure/azure-sdk-for-go/issues/16638